aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -93,18 +92,25 @@ class QLinearInt8 final : public torch::OperatorKernel {
     // This is the end of the pipeline, pass the resulting matrix through.
     fbgemm::DoNothing<> doNothingObj{};
 
+    at::Tensor bias = pack_ptr.bias;
+
     const int32_t* bias_ptr = nullptr;
-    if (bias.has_value()) {
-      Tensor bias_vec = bias.value();
-      TORCH_CHECK(bias_vec.dim() == 1, "bias should be a vector (1D Tensor)");
+
+    if (bias.defined()) {
+      // Temporary: Quantize bias
+      at::Tensor qbias = bias;
+      if (!bias.is_quantized()) {
+        qbias = at::quantize_linear(
```

Comment:
this is different logic from what was before this diff. Note that bias is already quantized, but with wrong scale/zp. Therefore, here if it's quantized we'd need to requantize it with correct scale.

Btw, if our unittests didn't catch it - we have crappy unittests.

Reply:
Dequantizing and re-quantizing the bias is causing various unit tests to fail, with either mismatching output or ASAN errors. I pinged @jianyuh to help with this. Also cc @dskhudia .

Reply:
For bias in `fbgemm_linear`, it is already quantized outside and the `bias` here should be `torch.qint32` data type. Check https://github.com/pytorch/pytorch/blob/master/test/test_quantized.py#L812-L813. So we don't need to apply `quantize_linear` inside this operator

Reply:
This is the same as the previous code in https://github.com/pytorch/pytorch/blob/master/torch/nn/quantized/modules/linear.py#L133, which always requantizes the bias. In the commented out code below, I'm trying to do that but it causes unit tests to fail which is very weird.

Reply:
This part of the code needs to handle per channel quantization. See use of qscheme above in the code. This should make your ASAN errors go away. 
`qbias = at::quantize_linear(at::dequantize(bias), pack_ptr.w_scale[0] * input_scale_float, 0, kQInt32);`

I still see some output mismatch even if this issue is not there. I am not yet sure why.

Reply:
@dskhudia good catch. I'll take that into account. However this fails even without per channel quantization enabled.

Reply:
Added code needed that check. Once we handle per channel case,  the check would be unnecessary.

Reply:
Please see my comment below for the other data mismatch issue.